packages/datetime/test/timePickerTests.tsx

```diff
@@ -209,6 +209,214 @@ describe("<TimePicker>", () => {
+            assert.strictEqual(millisecondInput.value, "200");
+            assert.strictEqual(timePicker.state.value.getMilliseconds(), 200);
+        });
+
+        it("when I typed time smaller than minTime, after blur, time should be reset to last good state", () => {
+            renderTimePicker({
+                defaultValue: new Date(1995, 6, 30, 15, 32, 20, 600),
+                minTime: new Date(1995, 6, 30, 15, 32, 20, 600),
+                precision: TimePickerPrecision.MILLISECOND,
+            });
+
+            const hourInput = findInputElement(Classes.TIMEPICKER_HOUR);
+            const minuteInput = findInputElement(Classes.TIMEPICKER_MINUTE);
+            const secondInput = findInputElement(Classes.TIMEPICKER_SECOND);
+            const millisecondsInput = findInputElement(Classes.TIMEPICKER_MILLISECOND);
+
+            changeInputThenBlur(hourInput, "14");
+            assert.strictEqual(hourInput.value, "15");
+            assert.strictEqual(timePicker.state.value.getHours(), 15);
+
+            changeInputThenBlur(minuteInput, "31");
+            assert.strictEqual(minuteInput.value, "32");
+            assert.strictEqual(timePicker.state.value.getMinutes(), 32);
+
+            changeInputThenBlur(secondInput, "19");
+            assert.strictEqual(secondInput.value, "20");
+            assert.strictEqual(timePicker.state.value.getSeconds(), 20);
+
+            changeInputThenBlur(millisecondsInput, "500");
+            assert.strictEqual(millisecondsInput.value, "600");
+            assert.strictEqual(timePicker.state.value.getMilliseconds(), 600);
+        });
+
+        it("when I typed time greater than maxTime, after blur, time should be reset to last good state", () => {
+            renderTimePicker({
+                defaultValue: new Date(1995, 6, 30, 15, 32, 20, 600),
+                maxTime: new Date(1995, 6, 30, 15, 32, 20, 600),
+                precision: TimePickerPrecision.MILLISECOND,
+            });
+
+            const hourInput = findInputElement(Classes.TIMEPICKER_HOUR);
+            const minuteInput = findInputElement(Classes.TIMEPICKER_MINUTE);
+            const secondInput = findInputElement(Classes.TIMEPICKER_SECOND);
+            const millisecondsInput = findInputElement(Classes.TIMEPICKER_MILLISECOND);
+
+            changeInputThenBlur(hourInput, "16");
+            assert.strictEqual(hourInput.value, "15");
+            assert.strictEqual(timePicker.state.value.getHours(), 15);
+
+            changeInputThenBlur(minuteInput, "33");
+            assert.strictEqual(minuteInput.value, "32");
+            assert.strictEqual(timePicker.state.value.getMinutes(), 32);
+
+            changeInputThenBlur(secondInput, "21");
+            assert.strictEqual(secondInput.value, "20");
+            assert.strictEqual(timePicker.state.value.getSeconds(), 20);
+
+            changeInputThenBlur(millisecondsInput, "700");
+            assert.strictEqual(millisecondsInput.value, "600");
```

Comment:
❌ _please_ don't test DOM element value like this--it's not the canonical source of truth in React. 

**copy the other tests in this file.** they all look at `state.value` and use a test helper function that you should be able to easily reuse in all your tests:

```
assertTimeIs(timePicker.state.value, 0, 0, 0, 0);
```